### PR TITLE
[IMP] website: make editor closer to WYSIWYG

### PR DESCRIPTION
**Pull request by DramixDW**
I'm removing most of the "DRAG BUILDING BLOCK HERE" placeholders and
displaying some blocks that were hidden in preview mode so that
we can reduce the differences between preview mode and the view once the
page is saved.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/static/src/scss/website.edit_mode.scss [hunk 1]
```diff
@@ -51,6 +51,12 @@ $-editor-messages-margin-x: 2%;
             &.oe_structure_small {
                 padding: 15px 0px;
             }
+            &.oe_structure_none{
+                height: 0px;    
+                border-style: none;
+                padding: 0px;
+                margin: 0px;
+            }
```
Comment (qsm-odoo) [inline addons/website/static/src/scss/website.edit_mode.scss]: Here you add style which says: "When a `oe_structure` asks to have an editor message displayed (`oe_empty`), remove it if it has `oe_structure_none`"... what would be the difference with removing `oe_empty` in those cases ?

File: addons/website/data/website_data.xml [hunk 2]
```diff
@@ -57,7 +57,7 @@
                                     </div>
                                 </div>
                             </div>
-                            <div class="oe_structure mt-2"/>
+                            <div class="oe_structure mt-2 oe_structure_none" data-editor-message=""/>
```
Comment (qsm-odoo) [inline addons/website/data/website_data.xml]: Clearly, we don't want to have to add `data-editor-message=""` everywhere, I think this is the opposite that we want.

File: addons/$ [hunk 3]
```diff
@@ -0,0 +1,1747 @@
...
+    </template>
+
+    <template id="assets_backend" inherit_id="web.assets_backend">
+        <xpath expr="." position="inside">
+            <script type="text/javascript" src="/website_sale/static/src/js/website_sale_video_field_preview.js"></script>
+            <script type="text/javascript" src="/website_sale/static/src/js/website_sale_backend.js"></script>
+            <link rel="stylesheet" type="text/scss" href="/website_sale/static/src/scss/website_sale_dashboard.scss"/>
+            <link rel="stylesheet" type="text/scss" href="/website_sale/static/src/scss/website_sale_backend.scss"/>
+            <script type="text/javascript" src="/website_sale/static/src/js/tours/website_sale_shop_backend.js"></script>
+        </xpath>
```
Comment (qsm-odoo) [inline addons/$:16]: What is that file..? You cannot say a task is finished without reviewing it yourself first. This is literally the first file which appears in the PR...

File: addons/website/views/snippets/snippets.xml [hunk 4]
```diff
@@ -550,10 +550,10 @@
...
     <div data-js="CoverProperties" data-selector=".o_record_cover_container" data-no-check="true">
         <we-imagepicker string="Image" data-background="" data-name="cover_background_opt"/>
-        <we-select string="Size" data-cover-opt-name="size">
-            <we-button data-select-class="o_record_has_cover o_full_screen_height">Full Screen</we-button>
-            <we-button class="o_record_cover_opt_size_default" data-select-class="o_record_has_cover o_half_screen_height">Half Screen</we-button>
-            <we-button data-select-class="o_record_has_cover cover_auto">Fit text</we-button>
+        <we-select string="Size" data-name="cover_background_size">
+            <we-button data-select-class="o_full_screen_height">Full Screen</we-button>
+            <we-button class="o_record_cover_opt_size_default" data-select-class="o_half_screen_height">Half Screen</we-button>
+            <we-button data-select-class="cover_auto">Fit text</we-button>
```
Comment (qsm-odoo) [inline addons/website/views/snippets/snippets.xml:556]: Can you explain those change to me? I feel it is breaking everything but there should be a reason I don't know?

**Comment by sbuhl** [comment]
Hello @qsm-odoo and @DramixDW 
We still want to have the message on an empty page: 
"On a new, empty, page or on your home page, it makes sense to have the message
saying the area is editable by dropping few building blocks to make a stunning website."

**Comment by qsm-odoo** [comment]
@DramixDW Adapt according to SBU's message and come back with a explanation of the effect of each oe_structure/oe_empty/whatever classes ;)

**Comment by qsm-odoo** [comment]
@DramixDW You say everyday that you are working on this but the branch has not been updated since 20 days... so what?

File: addons/event/models/event_event.py [hunk 5]
```diff
@@ -102,9 +102,18 @@ def _get_default_stage_id(self):
...
         return event_stages[0] if event_stages else False
 
+    def _default_description(self):
+        return '''
+            <section class="s_text_block" data-snippet="s_text_block">\
+                <h5>''' + _("Join us for this 24 hours Event") + '''</h5>
+                <p>''' + _("Every year we invite our community, partners and end-users to come and meet us! It's the ideal event to get together and present new features, roadmap of future versions, achievements of the software, workshops, training sessions, etc....\
+    This event is also an opportunity to showcase our partners' case studies, methodology or developments. Be there and see directly from the source the features of the new version!") + '''</p>\
+            </section>
+        '''
```
Comment (qsm-odoo) [inline addons/event/models/event_event.py:112]: Put that in a XML view which is rendered here.

File: addons/website/static/src/js/editor/snippets.options.js [hunk 6]
```diff
@@ -1928,10 +1929,8 @@ options.registry.CoverProperties = options.Class.extend({
      * @override
      */
     _computeWidgetVisibility: function (widgetName, params) {
-        const hasCover = this.$target.hasClass('o_record_has_cover');
         if (params.coverOptName) {
-            var notAllowed = (this.$target.data(`use_${params.coverOptName}`) !== 'True');
-            return (hasCover && !notAllowed);
+            return this.$target.data(`use_${params.coverOptName}`) === 'True';
         }
         return this._super(...arguments);
```
Comment (qsm-odoo) [inline addons/website/static/src/js/editor/snippets.options.js:1935]: Remind me what does that change do?

File: addons/website/static/src/scss/website.edit_mode.scss [hunk 7]
```diff
@@ -45,12 +45,14 @@ $-editor-messages-margin-x: 2%;
         opacity: 0.3;
     }
 
-    &.oe_structure.oe_empty, &[data-oe-type=html], .oe_structure.oe_empty {
-        &:empty {
+    &.oe_structure.oe_empty, .oe_structure.oe_empty {
+        /*Making the drop zone displayed on drag small*/
+        &.oe_structure_small > .oe_drop_zone.oe_insert:not(.oe_vertical):only-child {
```
Comment (qsm-odoo) [inline addons/website/static/src/scss/website.edit_mode.scss:50]: Why is there still this `oe_structure_small` addition? Did the specs change again?

File: addons/website/static/src/scss/website.edit_mode.scss [hunk 8]
```diff
@@ -45,12 +45,14 @@ $-editor-messages-margin-x: 2%;
...
+            padding: 12px 0px;
+        }
+        /*Display drop zone with placeholder on empty page*/
+        &:empty:only-child {
```
Comment (qsm-odoo) [inline addons/website/static/src/scss/website.edit_mode.scss:54]: Weak condition with `only-child`. Why not targeting `#wrap.oe_structure.oe_empty`?

File: addons/website_blog/static/src/scss/website_blog.scss [hunk 9]
```diff
@@ -139,54 +139,56 @@ $o-wblog-loader-size: 50px;
...
+                }
+
+                &.o_half_screen_height {
+                    min-height: 250px;
+                }
+
+                &.cover_auto {
+                    min-height: 150px;
+                    padding: 0;
+                }
```
Comment (qsm-odoo) [inline addons/website_blog/static/src/scss/website_blog.scss:171]: What does this have to do with wyswyg drop zone?

File: addons/website/static/src/js/editor/snippets.options.js (hunk 6 above)
Comment (DramixDW) [reply addons/website/static/src/js/editor/snippets.options.js:1935]: It makes the cover options available even if there's no background-image

File: addons/website/static/src/scss/website.edit_mode.scss (hunk 7 above)
Comment (DramixDW) [reply addons/website/static/src/scss/website.edit_mode.scss:50]: It's the class that make the zone only appears on drag. But i realize that my comment is wrong and the naming isn't clear

Comment (qsm-odoo) [reply addons/website/static/src/scss/website.edit_mode.scss:50]: > It's the class that make the zone only appears on drag.
But the spec is that we want that everywhere... what do we need to add a class to do it?

File: addons/event/models/event_event.py [hunk 10]
```diff
@@ -102,9 +102,12 @@ def _get_default_stage_id(self):
         event_stages = self.env['event.stage'].search([])
         return event_stages[0] if event_stages else False
 
+    def _default_description(self):
+        return self.env['ir.ui.view']._render_template('event.event_default_descripton')
```
Comment (qsm-odoo) [inline addons/event/models/event_event.py:106]: @JKE-be I would have written `self.env.ref('event.event_default_descripton').render_template()` or something like that but not sure? What about `viewref` when not in website?

Comment (JKE-be) [reply addons/event/models/event_event.py:106]: both are valid imo

self.env['ir.ui.view']._render_template('event.event_default_descripton')
==
self.env.ref('event.event_default_descripton')._render()

In all case, render became recently private and view are no mode readable for public/portal user. 

KR

File: addons/website/static/src/scss/website.edit_mode.scss [hunk 11]
```diff
@@ -45,12 +45,10 @@ $-editor-messages-margin-x: 2%;
         opacity: 0.3;
     }
 
-    &.oe_structure.oe_empty, &[data-oe-type=html], .oe_structure.oe_empty {
-        &:empty {
+    &.oe_structure.oe_empty, .oe_structure.oe_empty {
+        &#wrap:empty, &#wrap >.oe_drop_zone.oe_insert:not(.oe_vertical):only-child {
```
Comment (JKE-be) [inline addons/website/static/src/scss/website.edit_mode.scss:49]: it break the editor-message on eCommerce category ! 
Please double check !

Comment (qsm-odoo) [reply addons/website/static/src/scss/website.edit_mode.scss:49]: Probably because you removed `&[data-oe-type=html]` @DramixDW, do you know why you did?
